tests/filecheck/backend/rvscf_register_allocation.mlir

```diff
@@ -0,0 +1,115 @@
+// CHECK-NEXT:        %9 = riscv.add %8, %7 : (!riscv.reg<a3>, !riscv.reg<a0>) -> !riscv.reg<a3>
+// CHECK-NEXT:        "riscv_scf.yield"(%9) : (!riscv.reg<a3>) -> ()
+// CHECK-NEXT:      }) : (!riscv.reg<a0>, !riscv.reg<a1>, !riscv.reg<a2>, !riscv.reg<a3>) -> !riscv.reg<a3>
+// CHECK-NEXT:      %10 = riscv.mv %6 : (!riscv.reg<a3>) -> !riscv.reg<a0>
+// CHECK-NEXT:      riscv_func.return %10 : !riscv.reg<a0>
+// CHECK-NEXT:    }
+// CHECK-NEXT:  }
+
+// -----
+
+builtin.module {   
+  riscv_func.func @foo(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>) {
+    %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<>
+    %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<>
+    %4 = riscv.li 1 : () -> !riscv.reg<>
+    %55 = riscv.li 0 : () -> !riscv.reg<>
+    %5 = riscv.fcvt.s.w %55 : (!riscv.reg<>) -> !riscv.freg<>
+    %6 = "riscv_scf.for"(%2, %3, %4, %5) ({
+    ^1(%7 : !riscv.reg<>, %8 : !riscv.freg<>):
+      %99 = riscv.fcvt.s.w %7 : (!riscv.reg<>) -> !riscv.freg<>
+      %9 = riscv.fadd.s %8, %99 : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+      "riscv_scf.yield"(%9) : (!riscv.freg<>) -> ()
+    }) : (!riscv.reg<>, !riscv.reg<>, !riscv.reg<>, !riscv.freg<>) -> !riscv.freg<>
+    %10 = riscv.fcvt.w.s %6 : (!riscv.freg<>) -> !riscv.reg<a0>
+    "riscv_func.return"(%10) : (!riscv.reg<a0>) -> ()
+  }
+}
+
+// CHECK:       builtin.module {
+// CHECK-NEXT:    riscv_func.func @foo(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>) {
+// CHECK-NEXT:      %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+// CHECK-NEXT:      %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
+// CHECK-NEXT:      %4 = riscv.li 1 : () -> !riscv.reg<a2>
+// CHECK-NEXT:      %5 = riscv.li 0 : () -> !riscv.reg<a3>
+// CHECK-NEXT:      %6 = riscv.fcvt.s.w %5 : (!riscv.reg<a3>) -> !riscv.freg<fa0>
+// CHECK-NEXT:      %7 = "riscv_scf.for"(%2, %3, %4, %6) ({
+// CHECK-NEXT:      ^0(%8 : !riscv.reg<a0>, %9 : !riscv.freg<fa0>):
+// CHECK-NEXT:        %10 = riscv.fcvt.s.w %8 : (!riscv.reg<a0>) -> !riscv.freg<fa1>
+// CHECK-NEXT:        %11 = riscv.fadd.s %9, %10 : (!riscv.freg<fa0>, !riscv.freg<fa1>) -> !riscv.freg<fa0>
+// CHECK-NEXT:        "riscv_scf.yield"(%11) : (!riscv.freg<fa0>) -> ()
+// CHECK-NEXT:      }) : (!riscv.reg<a0>, !riscv.reg<a1>, !riscv.reg<a2>, !riscv.freg<fa0>) -> !riscv.freg<fa0>
+// CHECK-NEXT:      %12 = riscv.fcvt.w.s %7 : (!riscv.freg<fa0>) -> !riscv.reg<a0>
+// CHECK-NEXT:      riscv_func.return %12 : !riscv.reg<a0>
+// CHECK-NEXT:    }
+// CHECK-NEXT:  }
+
+// -----
+
+builtin.module {   
+  riscv_func.func @foo(%arg0 : !riscv.reg<a0>) {
+    %0 = riscv.li 0 : () -> !riscv.reg<>
+    %2 = riscv.li 0 : () -> !riscv.reg<>
+    %3 = riscv.li 1 : () -> !riscv.reg<>
+    %5 = "riscv_scf.for"(%2, %arg0, %3, %0) ({
+    ^1(%arg1 : !riscv.reg<>, %arg2 : !riscv.reg<>):
+      %6 = riscv.li 0 : () -> !riscv.reg<>
+      %7 = riscv.li 1 : () -> !riscv.reg<>
+      %9 = "riscv_scf.for"(%6, %arg0, %7, %arg2) ({
+      ^2(%arg3 : !riscv.reg<>, %arg4 : !riscv.reg<>):
```

Comment:
I find this `^1` and `^2` (xdsl) still a bit confusing, since technically they are both their regions `^bb0`(mlir), right?

Reply:
these are basic blocks not regions, but yeah they could have the same name. I think we would do well to add custom syntax here, as then we wouldn't need the basic block argument printing.

Reply:
Yes I meant basic blocks, sorry.
But I agree this is not for this PR, either by fixing the way xdsl prints BB's or indeed by adding custom syntax :wink: